Use events API

asermax opened the pull request:
This PR replaces the usage of the `events.json` file for an API. SWR was added to make the request, which currently relies on using the current month and year to only fetch events for the time range showed on the calendar.
Evntually when #148 gets implemented, the API also provides a `upcoming` filter that only returns upcoming events.

homer0 commented:
Sorry I didn't have enough time to research it, but there seems to be a tiny issue when you change months the first time: the request for the new month is triggered.... after a request for the month that you are currently seeing :P.

Btw, how did you skip CORs?

joelalejandro commented:
@homer0 I see CORS headers on the AWS lambda:

``` 
access-control-allow-origin: https://codear-site-git-feat-events-api.codear.vercel.app
```

and on the request as well:

```
sec-fetch-mode: cors
```

asermax commented:
Yup, I ended up adding cors support on the django app and added some exceptions for vercel urls + codear.org.
Regarding the double query issue, I'll give it a look later today; maybe it has to do with SWR but I'm not sure. In any case, I also added `cache-control` headers (I think they last for a day) to avoid a lot of unnecessary requests.

asermax commented:
@homer0 I'm not 100% sure but this seems to be a Schrödinger issue: SWR, by default, revalidates the data when the browser gains focus; whenever you open the dev tools and go to the network tab you're taking the focus away from the browser and then, when you click on the arrow to go to the next month, you're giving it focus again, which causes a revalidation followed by the fetch of the following month. So basically, the bug happens because you're trying to find it!
